control-plane/subcommand/common/metrics_util.go

```diff
@@ -0,0 +1,32 @@
+package common
+
+import "strconv"
+
+func ParseScrapePort(v string) (int, bool) {
+	port, err := strconv.Atoi(v)
+	if err != nil {
+		// we only use the port if it's actually valid
+		return 0, false
+	}
+	if port < 1024 || port > 65535 {
+		return 0, false
+	}
+	return port, true
+}
+
+func GetScrapePath(v string) (string, bool) {
+	if v == "" {
+		return "", false
+	}
+	return v, true
```

Comment:
I think this is equivalent

```suggestion
	return v, v != ""
```